crest/Assets/Crest/Crest/Scripts/Collision/SamplingHelpers.cs

```diff
@@ -10,12 +12,32 @@ namespace Crest
     /// Helper to obtain the ocean surface height at a single location per frame. This is not particularly efficient to sample a single height,
     /// but is a fairly common case.
     /// </summary>
+#if CREST_BURST_QUERY
+    [BurstCompile]
+#endif
     public class SampleHeightHelper
     {
+#if CREST_BURST_QUERY
+        Vector3 _queryPos;
+        Vector3 _queryResult;
+        Vector3 _queryResultNormal;
+        Vector3 _queryResultVel;
+
+        // These are only ever used on the main thread, and they're filled in when they're used, so we only need one
+        // global copy of them. if we had one for every sample height helper, we would spend forever creating and
+        // destroying nativearrays. consider revisiting this when 22.2 is the minspec, because then nativearray becomes
+        // fully unmanaged. Also, obviously revisit if their usage is ever jobified.
```

Comment:
Not sure about this as when would we ever be creating/destroying this helper class frequently? Happy to go with it though.

Reply:
The solution in this comment does save users from disposing this class. May leave comment as-is.